apiv6/kentikapi/httputil/httputil_test.go

```diff
@@ -106,6 +107,48 @@ func TestRetryingClientWithSpyHTTPTransport_Do(t *testing.T) {
 	}
 }
 
+func TestRetryAfter(t *testing.T) {
+	var (
+		header     string
+		httpErrors = []int{
+			http.StatusBadGateway,
+			http.StatusTooManyRequests,
+			http.StatusTooManyRequests,
+			http.StatusBadRequest,
+		}
+		retryAfterValue = 3 * time.Second
+	)
+
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		header = time.Now().Add(retryAfterValue).Format(time.RFC1123)
+		w.Header().Set("Retry-After", header)
+		http.Error(w, "", httpErrors[0])
+		httpErrors = httpErrors[1:]
+	}))
+	defer ts.Close()
+
+	c := httputil.NewRetryingClient(httputil.ClientConfig{
+		RetryCfg: httputil.RetryConfig{
+			MaxAttempts: intPtr(5),
+			MinDelay:    durationPtr(1 * time.Second),
+			MaxDelay:    durationPtr(10 * time.Second),
+		},
+	})
+
+	req, err := retryablehttp.NewRequest(http.MethodGet, ts.URL, nil)
+	require.NoError(t, err)
+
+	// act
+	//nolint:bodyclose
+	resp, err := c.Do(req.WithContext(context.Background()))
+
+	// assert
```

Comment:
This test does not verify that duration of retry-after is used by retry mechanism.
I propose following modifications:

1. httpErrors to contain only retry-after-compatible statuses, except last one: [StatusServiceUnavailable, StatusTooManyRequests, StatusOk]
2. Set both MinDelay and MaxDelay to large numbers, e.g. 10 Seconds.
3. Set context.Timeout passed to c.Do() to e.g. 3 Seconds.
4. Set retryAfterValue to low value e.g. 10 Milliseconds.
5. Verify that there is no error returned from c.Do() and no error in context (there was no timeout).

Reply:
The problem with low value for retryAfterValue is the format of RFC1123. `RFC1123     = "Mon, 02 Jan 2006 15:04:05 MST"`
This only takes into account the seconds part. The way this works when I run the same test with e.g. retry-after header 500 ms is it sometimes retries after random amount of time between 0ms and 250-ish ms but majority of time it's 0ms because time.Now() + 500 ms is approximated to full second and usually already in the past when server reads when to retry. 
I suggest 1 second for retry-after or - if that's too long - complete restructuring of the test.

Reply:
Ok, you can use 1 second for retryAfterValue then.
Using 10 ms wouldn't be too bad either, because it would still test that the code respects the retry-after header, which is the most important thing here.

Reply:
Done